ansible_wisdom/ai/api/tests/test_views.py

```diff
@@ -266,6 +266,31 @@ def test_wca_completion(self):
             model_client.session.post.call_args.kwargs["json"]["model_id"], "org-model-id"
         )
 
+    @override_settings(WCA_SECRET_DUMMY_SECRETS="1:valid")
+    @override_settings(ENABLE_ADDITIONAL_CONTEXT=True)
+    @override_settings(ENABLE_ARI_POSTPROCESS=False)
+    def test_wca_completion_anonymized_additional_context(self):
+        self.user.rh_user_has_seat = True
+        self.user.organization = Organization.objects.get_or_create(id=1)[0]
+        self.client.force_authenticate(user=self.user)
+
+        stub = self.stub_wca_client(
+            200,
+        )
+        model_client, model_input = stub
+        model_input["metadata"] = {
+            "additionalContext": {
+                "userid": "a-user-id",
+                "password": "a-password",
+            },
+        }
+        self.mock_model_client_with(model_client)
+        r = self.client.post(reverse("completions"), model_input, format="json")
+        self.assertEqual(r.status_code, HTTPStatus.OK)
+        self.assertEqual(
```

Comment:
@robinbobbitt I need to assert that WCA is called with the additional context...

But this is where I don't understand _how_ WCA is meant to be passed the additional context.

I can trace passage of the `metadata` containing the additional context from the HTTP `request` into `model_mesh_client.infer(...)` but at this point `metadata` is dropped and only the `prompt` and `context` passed on (to WCA). I'm guessing something _extra_ needs adding.. but I'm not sure _how_ we should pass additional context to WCA. 

Can you advise?

Reply:
I worked something out. See `test_views.py`.